common/src/main/java/com/wynntils/services/mapdata/MapDataService.java

```diff
@@ -55,15 +59,27 @@ public class MapDataService extends Service {
     private final Map<MapFeature, ResolvedMapAttributes> resolvedAttributesCache = new HashMap<>();
     private final Map<String, Optional<MapIcon>> iconCache = new HashMap<>();
 
+    // Storage for json providers
+    @Persisted
```

Comment:
Conceptually this should be a `HiddenConfig` instead?

Reply:
I am not sure. I thought we wanted to move waypoints out of configs. At least, that's what I remember. I am not even sure where normal waypoints are stored in mapdata.

Reply:
I think there is a difference between storing users waypoint, and storing providers. This is, afaict, the latter, and I think that has a much more "config-y" feel to it.

Reply:
But the huge chunk of detected loot chests or added waypoints are not. It's not a configuration, as much as incorporating online providers is. I think.

Reply:
Sadly this means I have to implement a `onStorageLoad` variant for configs for all core components.

Reply:
Didn't we both forget that configs don't work with core components? I only got reminded when looking at the annoyingly similar code of the two, where storage owners implement a nice Storageable interface and config ownership is merged into Feature / PersistedValue / PersistedMetadata itself.